### PR TITLE
[meta] update to Rust 1.80

### PR DESCRIPTION
**Pull request by sunshowers**
The most interesting change here is that Rust now detects unused cfg flags.

* `sp-sim` referred to `no-caboose` and `old-state` features that no longer exist. Per @labbott this code is okay to remove.
* In `omicron-dev`, `cfg(mac)` is not valid -- replaced it with `target_os = "macos"`.

There were also a few new clippy warnings.

### CODE REVIEW

**Comment by sunshowers** [comment]
Ah need to fix the illumos build.

File: oximeter/impl/src/histogram.rs [hunk 1]
```diff
@@ -412,6 +412,14 @@ macro_rules! impl_int_sample {
...
                     // For squared mean (M2) calculation, before we update the
                     // count.
+                    //
+                    // Some of these casts are lossless (i8, u8 etc) and some
+                    // are not (i64, u64). The point of the `cast_lossless`
+                    // lint is to ensure that if the type changes out from
+                    // underneath, this fails to compile. Since we explicitly
+                    // know which types we're casting from, that concern is
+                    // moot.
+                    #[allow(clippy::cast_lossless)]
```
Comment (zeeshanlakhani) [inline oximeter/impl/src/histogram.rs:422]: exactly. Great comment.

**Comment by sunshowers** [comment]
Need to update helios-build it seems.

**Comment by sunshowers** [comment]
Need to wait for https://github.com/oxidecomputer/helios/pull/167 to land.

**Comment by sunshowers** [comment]
ah sorry, also need to update the helios hash in here I guess!

**Comment by sunshowers** [comment]
Actually no, I think we need to update one of the downstream deps. ugh time :(

**Comment by sunshowers** [comment]
A couple more time dependency updates required:

* https://github.com/oxidecomputer/pilot/pull/40
* https://github.com/illumos/image-builder/pull/7

**Review by jgallagher — APPROVED** [review]

**Review by zeeshanlakhani — APPROVED** [review]

**Pull request merged**